### PR TITLE
Fix Mizar crash on no frame track available

### PR DESCRIPTION
**Pull request by akopich**
If a capture has no frame track, an error message is shown on
startup and the `Update` button is disabled.

Test: Manual
Bug: http://b/240671648

### CODE REVIEW

File: src/MizarWidgets/SamplingWithFrameTrackWidget.cpp [hunk 1]
```diff
@@ -100,4 +113,19 @@ void SamplingWithFrameTrackWidget::OnUpdateButtonClicked() {
   ui_->output_->UpdateReport(std::move(report));
 }
 
+bool SamplingWithFrameTrackWidget::IsDataValid(const orbit_mizar_data::MizarPairedData& data,
+                                               std::string_view data_title) const {
+  const bool is_valid = !data.GetFrameTracks().empty();
+  if (!is_valid) {
+    emit ReportError(absl::StrCat(data_title,
```
Comment (florian-kuebler) [inline src/MizarWidgets/SamplingWithFrameTrackWidget.cpp:120]: I am not too sure about the right design here. Should we emit the error report here, or return an ErrorMessageOr and handle it later?

Anyways. If we emit the error here, I would rename the function, to e.g. IsDataValidAndEmitErrorIfNot

File: src/MizarWidgets/SamplingWithFrameTrackWidget.cpp [hunk 2]
```diff
@@ -47,6 +51,15 @@ SamplingWithFrameTrackWidget::SamplingWithFrameTrackWidget(QWidget* parent)
 void SamplingWithFrameTrackWidget::Init(
     const orbit_mizar_data::BaselineAndComparison* baseline_and_comparison,
     const Baseline<QString>& baseline_file_name, const Comparison<QString>& comparison_file_name) {
+  const auto validator = absl::bind_front(&SamplingWithFrameTrackWidget::IsDataValid, this);
```
Comment (florian-kuebler) [inline src/MizarWidgets/SamplingWithFrameTrackWidget.cpp:54]:
```suggestion
  const auto validate = absl::bind_front(&SamplingWithFrameTrackWidget::IsDataValid, this);
```

Comment (florian-kuebler) [reply src/MizarWidgets/SamplingWithFrameTrackWidget.cpp:54]: Or validate_function

File: src/MizarWidgets/SamplingWithFrameTrackWidget.cpp (hunk 1 above)
Comment (akopich) [reply src/MizarWidgets/SamplingWithFrameTrackWidget.cpp]: Good point. Thank you. The function returns ErrorMessageOr now.

File: src/MizarWidgets/SamplingWithFrameTrackWidget.cpp (hunk 2 above)
Comment (akopich) [reply src/MizarWidgets/SamplingWithFrameTrackWidget.cpp]: Removed the bind_front.

**Review by florian-kuebler — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**